realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java

```diff
@@ -173,9 +160,25 @@ private void emitAccessors(JavaWriter writer) throws IOException {
                 writer.emitStatement(
                         "realm.checkIfValid()"
                 );
+
+                // For String and bytes[], null value will be returned by JNI code. Try to save one JNI call here.
+                if (metadata.isNullable(field) && !Utils.isString(field) && !Utils.isByteArray(field)) {
+                    writer.beginControlFlow("if (row.isNull(%s))", staticFieldIndexVarName(field));
```

Comment:
For  non-null String, it would call `row.isNull()` and then `row.getString()`, which are two jni calls for all non-null cases. And here is just call `row.getString()` for all cases, which saves one jni call for all non-null String cases.